wordpress-mobile/WordPress-iOS
# Issues/4094 reader menu

Refs #4094 
In the interest of smaller PRs, this is the first stage of the reader menu refresh. It includes:
- A feature flag to preserve the original menu until the new one is ready
- A new view model to decouple the menu from core data. This allows arbitrary items to be inserted among items persisted in core data. 
- A new menu controller with basic presentation functionality. 

Future PRs will reintroduce tag and following management. 

To test: 
- Confirm that the logged in and logged out menus are presented and have the correct items. 
- Tap the different items in the menu and ensure that each type of content loads correctly. 

Needs review: @jleandroperez would you take a peek?
